### PR TITLE
PST-006-create-button-component-PST-007-create-add-new-entry-modal

### PR DESCRIPTION
**Pull request by skrzypczyk-pawel**

### CODE REVIEW

File: src/Modals/BaseModal.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import React, { FC, Dispatch, SetStateAction } from "react";
+
+import { CloseIcon } from "src/assets/icons";
+
+import { css, StyleSheet } from "aphrodite";
+import { colors } from "src/styles";
+
+interface Props {
+  show: boolean;
```
Comment (Stared55) [inline src/Modals/BaseModal.tsx]: visible

File: src/Modals/BaseModal.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  callback: Dispatch<SetStateAction<boolean>>;
```
Comment (Stared55) [inline src/Modals/BaseModal.tsx]: close: () => void

File: src/Modals/BaseModal.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+export const BaseModal: FC<Props> = ({ children, show, callback }) => {
+  const hideModal = () => {
+    return callback(!show);
+  };
+
+  return (
+    <div
+      onClick={hideModal}
```
Comment (Stared55) [inline src/Modals/BaseModal.tsx]: onClick={close}

File: src/Modals/BaseModal.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      className={css(styles.backdrop, show && styles.hide)}
```
Comment (Stared55) [inline src/Modals/BaseModal.tsx:21]: visible && styles.visible

File: src/Modals/BaseModal.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    >
+      <div className={css(styles.modal)} onClick={(e) => e.stopPropagation()}>
+        {children}
+        <button
+          type="button"
+          className={css(styles.button)}
+          onClick={hideModal}
```
Comment (Stared55) [inline src/Modals/BaseModal.tsx]: we need to do it together

File: src/components/AddNewTodoForm.tsx [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+import { StyledInput } from "./StyledInput";
+import { TextArea } from "./TextArea";
+import { Categories } from "./Categories";
+import { SelectMenu } from "./SelectMenu";
+import { StyledOption } from "./StyledOption";
+import { StyledText } from "./StyledText";
+
+const AddNewTodoSchema = Yup.object().shape({
+  title: Yup.string()
+    .min(3, "Too Short!")
```
Comment (Stared55) [inline src/components/AddNewTodoForm.tsx]: i18n

File: src/components/AddNewTodoForm.tsx [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+  description: Yup.string()
+    .min(5, "Too Short!")
+    .max(60, "Too Long!")
+    .required("Required"),
+});
+
+interface Props {}
+
+export const AddNewTodoForm: FC<Props> = () => {
+  const selectedMenu = document.getElementById(
```
Comment (Stared55) [inline src/components/AddNewTodoForm.tsx:30]: You need to explain me this const

File: src/components/AddNewTodoForm.tsx [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+    "formSelectMenu"
+  ) as HTMLSelectElement;
+  const [category, setCategory] = useState<Category>("none");
+  const [priority, setPriority] = useState<Priority>("low");
+  const getPriority = () => {
+    const getImportance = selectedMenu?.value;
```
Comment (Stared55) [inline src/components/AddNewTodoForm.tsx:36]: same as above

File: src/components/AddNewTodoForm.tsx [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+        return setPriority("low");
+      default:
+        return setPriority("low");
+    }
+  };
+
+  return (
+    <>
+      <Formik
+        initialValues={{
```
Comment (Stared55) [inline src/components/AddNewTodoForm.tsx]: make `const initialValues` above the component

File: src/components/AddNewTodoForm.tsx [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+          title: "",
+          status: "todo",
+          priority,
+          category,
+          deadline: "01-01-2034",
+          description: "",
+          createdAt: new Date().toISOString(),
+        }}
+        validationSchema={AddNewTodoSchema}
+        onSubmit={(values, actions) => {
```
Comment (Stared55) [inline src/components/AddNewTodoForm.tsx]: make a new function `handleOnSubmit`

File: src/components/AddNewTodoForm.tsx [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+                <CircleIconButton
+                  iconName="health-icon"
+                  onClick={() => setCategory("health")}
+                  title={i18n.t("modal:healthCategory")}
+                  style={styles.categoryButton}
+                  isSelected={category === "health"}
+                />
+              </Categories>
+              <SelectMenu id="formSelectMenu">
+                <StyledOption value={0} text={i18n.t("modal:howImportant")} />
```
Comment (Stared55) [inline src/components/AddNewTodoForm.tsx]: I guess, you can make it more readable like `value='low'`

File: src/components/BaseModal.tsx [hunk 12]
```diff
@@ -0,0 +1,81 @@
+import React, { FC } from "react";
+
+import { css, StyleSheet } from "aphrodite";
+import { colors, getShadow } from "src/styles";
+import { CircleIconButton } from "./CircleIconButton";
+
+interface Props {
+  visible: boolean;
+  handleCloseModal: () => void;
```
Comment (Stared55) [inline src/components/BaseModal.tsx]: closeModal is enough :D

File: src/components/AddNewTodoForm.tsx [hunk 13]
```diff
@@ -124,13 +136,16 @@ export const AddNewTodoForm: FC<Props> = () => {
                 />
               </Categories>
               <SelectMenu id="formSelectMenu">
-                <StyledOption value={0} text={i18n.t("modal:howImportant")} />
-                <StyledOption value={1} text={i18n.t("modal:veryImportant")} />
+                <StyledOption value="low" text={i18n.t("modal:howImportant")} />
+                <StyledOption
+                  value="high"
+                  text={i18n.t("modal:veryImportant")}
```
Comment (Stared55) [inline src/components/AddNewTodoForm.tsx]: this translation isn't related to modal so key should be smth like 'addTodo:veryImportant'

**Review by MRadek55 — APPROVED** [review]

**Pull request merged**